### ISSUE TITLE
Document Control Allocation concept for Developers

### ISSUE THREAD

**Issue by junwoo091400** [PX4/PX4-Autopilot#19939]
## Background
There were [some interest](https://discuss.px4.io/t/px4-dev-call-july-20-2022/28275#q2-custom-multicopter-control-using-dynamic-control-allocation-salvatore-9) in a documentation of how to write your custom Control Allocation Actuator effectiveness matrix & create a custom controller out of it, which I think is quite important detail that users / researchers should have access to.

Also, the Control Allocation pipeline isn't so straightforward for a regular developer / user, as it requires some understanding in math and control theory (matrix part). So it would be nice to document exactly how the `vehicle_torque_setpoint`, `actuator_outputs`, `actuator_controls`, etc are related to each other.

## Describe your preferred solution
1. Document the whole control allocation concept in theoretical level (Actuator Effectiveness matrix, Matrix inverse)
2. Document how control allocation pipeline is implemented in PX4 (All the meanings of the related uORB topics, which topics gets published where and gets processed where, and where the matrix inversion occurs, etc)
3. Document how to create your own custom Control Allocation model (Using a simple example, like any vehicle that isn't supported yet - @sfuhrer any ideas?)

## Additional context
There's already quite some [nice improvements](https://github.com/PX4/PX4-user_guide/pull/1962) coming in documenting how to setup / use control allocation, we just need a more in-depth documentation on how it works so that we can make developers and researchers' life easier!

**Comment by junwoo091400** [comment]
Basic support was added, implementing Control Allocation model should be done by referring to the latest code changes, so that part is unnecessary.

https://docs.px4.io/main/en/concept/control_allocation.html
https://github.com/PX4/PX4-user_guide/pull/1923

**Comment by rosiakpiotr** [comment]
Hey, correct me if I'm wrong but there is no documentation on the theoretical part You listed as first point in preferred solution - maybe it is available somewhere but making it available from the docs themselves would be very helpful, even in form of research paper reference. I found this issue to be most relevant thus commenting regards its closed state.

**Comment by sfuhrer** [comment]
Does https://www.youtube.com/watch?v=NQwepGLMzZ4 help you?

**Comment by rosiakpiotr** [comment]
> Does https://www.youtube.com/watch?v=NQwepGLMzZ4 help you?

Thank you, I watched it before posting my comment. I made substantial progress towards proper implementation my experimental feature by reading source code and working it from there. My personal desire for this documentation diluted as I learnt from the source code but for future researches I believe a block diagram with description how each part of `ControlAllocator` works with each other would be great.
